### DISCUSSION TITLE
Subquery using json datasource

### DISCUSSION THREAD

**Discussion by gabchx** [grafana/grafana-infinity-datasource#900] [Specific APIs]
Hi,
Thanks for the project, I was an early adopter and it has always serve my needs way better than anything else.
I take time to come to you today for asking a question about SUB HTTP query.

I wanted to know if there was a way to query a data-source using result from another query.
Let me give an example.
Let say our json rest api has an endpoint to get a list of cats born on a given time range each with it's **cat_id**.
> api/cats?from=yesterday&to=now

A other endpoint has **the color of a cat given it's cat_id**

> api/color/{cat_id}

How can I successful manage to get all cats and  their colors for a given time rang (${__from}, ${__to})?

![image](https://github.com/grafana/grafana-infinity-datasource/assets/83697171/13e7ddfd-dd30-456d-b502-94be1b652c84)

**Comment by yesoreyeram** [comment]
Infinity doesn't have a concept called subquery. instead what you can do is create a variable to return the cat ids and then use it with infinity pagination feature. (note: infinity pagination is limited to max 5 requests at the moment)
